ext/ddtrace_profiling_native_extension/stack_recorder.c

```diff
@@ -616,6 +741,17 @@ static VALUE _native_record_endpoint(DDTRACE_UNUSED VALUE _self, VALUE recorder_
   return Qtrue;
 }
 
+static VALUE _native_track_obj_allocation(DDTRACE_UNUSED VALUE _self, VALUE recorder_instance, VALUE new_obj, VALUE weight) {
+  ENFORCE_TYPE(weight, T_FIXNUM);
+  track_object(recorder_instance, new_obj, NUM2UINT(weight));
+  return Qtrue;
+}
```

Comment:
Since this is supposed to only be using for testing, I thought the flexibility of being able to pass arbitrary values could be interesting to check some of the safety features/excludes we could add in the future when we start asking for the heap size of the object?

Reply:
Ah, yeah, that sounds good -- I was thinking "we may not support these things so perhaps it's better to not allow them here", but if the approach is to "let's actually make sure to have this tested" it's even better :)